### PR TITLE
Sidebar

### LINKED ISSUE #51: sidebar Frontend
**Description of the issue**
Design of the sidebar component

**Files that need to be updated**
Frontend/src/components/sidebar.js
Frontend/src/components/sidebar.css

**Link**
(https://github.com/UNLV-CS472-672/2024-F-GROUP3-RebelHub/pull/33)

### PR DESCRIPTION
**Pull request by leed85**
Please complete the checklist so the group can quickly understand your PR.

Category
 X Feature
    Bug Fix
    Testing
    Documentation
    Other
Domain
X Frontend
   Backend
   Both

Description
Adds the frontend for the sidebar component

Related Issue(s)
profile page

New Additions
frontend/src/components -files added sidebar.js and sidebar.css
fronted/public folder - pictures

Modifications
should not have modified any existing files (if any are touched, it is an error)

New Dependencies
N/A

Visuals
N/A

Architecture of Your Changes
Addition of Sidebar

Usage and Integration
Permanent existing sidebar for some pages

Testing
N/A
design for sidebar

### CODE REVIEW

File: frontend/src/components/sidebar.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import './sidebar.css'
+import Link from 'next/link'
+import Image from 'next/image'
```
Comment (thecatfour) [inline frontend/src/components/sidebar.js]: I am curious why you are importing Image when you use img to display the logo.

File: frontend/src/components/sidebar.css [hunk 2]
```diff
@@ -0,0 +1,33 @@
+.sidebar {
+    width: 250px;
+    background-color: #1f1f1f;
```
Comment (barrioskevin) [inline frontend/src/components/sidebar.css]: can you make transparent.

File: frontend/src/components/sidebar.js (hunk 1 above)
Comment (leed85) [reply frontend/src/components/sidebar.js]: This was from left over from previous implementation, Ill update

File: frontend/src/components/sidebar.css (hunk 2 above)
Comment (leed85) [reply frontend/src/components/sidebar.css]: Ill update it right now

**Review by thecatfour — APPROVED** [review]
Necessary changes have been made. Now, all that needs to be solved is the merge conflicts in .gitignore.

File: frontend/src/components/sidebar.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+                    width={250} 
+                    height={100} 
+                    src="/UNLV.png"
+                    className='logo'
+                />
+            </div>
+            <Link href="https://www.youtube.com/watch?v=dQw4w9WgXcQ" className='link'>Home</Link>
+            <Link href="/profile" className='link'>Explore</Link>
+            <Link href="/profile" className='link'>Hubs</Link>
+            <Link href="/profile" className='link'>Messages</Link>
```
Comment (nnguyen144) [inline frontend/src/components/sidebar.js]: Messages, notifications, home, and profile do not have to be included as they will be included in the header navigation bar. You can also start changing some of the hrefs for existing pages (such as the calendar page /calendar)

**Review by nnguyen144 — CHANGES_REQUESTED** [review]

File: frontend/src/components/sidebar.js (hunk 3 above)
Comment (nnguyen144) [reply frontend/src/components/sidebar.js]: After the above have been removed, then this can be merged.

**Review by nnguyen144 — APPROVED** [review]

**Pull request merged**